PYTHON/activityLog/activityLog.py

```diff
@@ -0,0 +1,356 @@
+			self.mysql_usr, self.mysql_pass)
+		return self.run_command(cmd)[0].decode('utf-8')
+
+	def createTar(self, actLogDir = "~/activityLog"):
+		logcation = "{}/{}".format(actLogDir, self.logfilename)
+		tarlogcation = "{}/{}.tar.gz".format(actLogDir,self.logfilename)
+		try:
+			with tarfile.open(tarlogcation, "w:gz") as tar:
+				tar.add(logcation)
+			os.remove(logcation)
+		except Exception as e:
+			if self.silent_mode == False:
+				print("Error creating tar.gz file.  Exception {}.".format(e))
+
+	def writeTheLog(self, actLogDir = "~/activityLog"):
+		timestamp = self.dateString()
+		actLogDir = self.createLogDir()
+		self.createLogDir(actLogDir)
+		logcation = "{}/{}".format(actLogDir,self.logfilename)
+		try:
+			with open(logcation, 'w') as logfile:
+				logfile.write(self.getTopOutput())
+				logfile.write("\n\n\n\n\n\n\n")
+				logfile.write("Thoroughput on NetWork Interfaces:\n")
+				logfile.write(self.getNDeviceThroughput())
+				logfile.write("\n\n")
+				logfile.write("Daemons and Open Ports list:\n")
+				logfile.write(self.getDaemonsAndPorts())
+				logfile.write("\n\n")
+				logfile.write("Network Connections:\n")
+				logfile.write("\n\n\n\n\n\n\n")
+				# Insert the MySQL bit here
+				if self.mysql_usr != None:
+					for i in range(self.mysql_queries):
+						logfile.write("MySQL Queries Active at {}".format(timestamp))
+						logfile.write("\n")
+						logfile.write(self.sampleMySQL())
+						logfile.write("\n\n")
+						time.sleep(self.mysql_interval)
+		except Exception as e:
+			if self.silent_mode == False:
+				print("Error writing to log file. Exception: {}".format(e))
+
+		# Call createTar() and archive this bitch
+		self.createTar(actLogDir)
+
+
+def main():
+	### Create argument parameters. 
+	# TODO: Find a way to make it case-insensitive.  Something about type = str.lower, 
+	# but I dont know where.
+	parser = argparse.ArgumentParser(
+		description='Creates a general \'timeslice\' snapshot of activity on a server.')
+	parser.add_argument('--epoch', help='Prefix log filename with current epoch.', 
+		action="store_true")
+	parser.add_argument('--rootcheck', 
+		help='Checks if root filesystem is r/w and includes it in the log filename.'
+		' Typically requires root access to run.', 
+		action="store_true")
+	parser.add_argument('--mysql_usr', 
+		help='MySQL username for localhost. Activates MySQL state logging.')
+	parser.add_argument('--mysql_pwd', 
+		help='MySQL password for --mysql_usr.',
+		default = "")
+	parser.add_argument('--mysql_queries', 
+		help='Number of MySQL queries for each report. Default = 60.', default = 60)
+	parser.add_argument('--mysql_interval', 
+		help='Delay between MySQL queries, in seconds. Default = 0.25.', default = 0.25)
+	args=parser.parse_args()
+	# Convert args namespace to dictionary
+	args = vars(args)
+	# Initiate the ActivityLogger class object
+	activitylog = ActivityLogger()
+	# Populate activitylog variables from args.  Must be a smarter way to do this.
```

Comment:
This is a pretty common problem.  If you match up the names of the object attributes and the parameters (which you do except for `mysql_pass` vs `mysql_pwd`) you can do it with something like: 
```
for attr in ('epoch', 'rootcheck', 'mysql_usr', 'mysql_pass', 'mysql_queries', 'mysql_interval'): 
    setattr(activitylog, attr, args.get(attr))
```
That can blow up if your `ActivityLogger` class def gets out of sync with your argument list.   If you want to get more clever, you can, but the method above is pretty standard.